### PR TITLE
ISSD-3168 [PRM] Save pagination on PRM

### PR DESCRIPTION
**Pull request by abfranca**

### CODE REVIEW

**Comment by abfranca** [comment]
ci:test:sf

**Comment by abfranca** [comment]
ci:test:sf

**Comment by abfranca** [comment]
ci:test:sf

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/MDFRequestList.tsx [hunk 1]
```diff
@@ -56,11 +56,24 @@ const MDFRequestList = () => {
...
+		new URLSearchParams(window.location.href.split('?')[1])
+	);
+	const urlParamsEntries = urlParams.entries();
+	useEffect(() => {
+		window.history.replaceState(
+			null,
+			'',
+			`${Liferay.ThemeDisplay.getLayoutRelativeURL()}?${urlParams.toString()}`
+		);
+	}, [urlParams, urlParamsEntries]);
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/MDFRequestList.tsx:69]: Create an useQueryParams

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/hooks/useFilters.tsx [hunk 2]
```diff
@@ -13,13 +13,40 @@ import getActivityPeriodFilterTerm from '../utils/getActivityPeriodFilterTerm';
...
+		if (urlParams.get('searchterm')) {
+			initialFilter.searchTerm = urlParams.get('searchterm')!;
+		}
+
+		if (urlParams.getAll('status').length) {
+			initialFilter.status.value = urlParams.getAll('status')!;
+		}
+
+		return initialFilter;
+	});
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/hooks/useFilters.tsx:49]: Maybe you can create a callback for this

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/MDFRequestList.tsx [hunk 3]
```diff
@@ -56,11 +56,24 @@ const MDFRequestList = () => {
 	const {userAccount} = useDynamicFieldEntries();
 	const actions = usePermissionActions(ObjectActionName.MDF_REQUEST);
 
+	const [urlParams] = useState(
+		new URLSearchParams(window.location.href.split('?')[1])
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/MDFRequestList.tsx:60]: window.location.search

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/usePagination.tsx [hunk 4]
```diff
@@ -5,9 +5,22 @@
...
+
 	const [activeDelta, setActiveDelta] = useState<number>(20);
-	const [activePage, setActivePage] = useState<number>(1);
+	const [activePage, setActivePage] = useState<number>(
+		urlParams ? Number(urlParams.get('activepage')) : 1
+	);
+
+	const handlePageChange = (newPage: number) => {
+		if (urlParams) {
+			urlParams.set('activepage', `${newPage}`);
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/usePagination.tsx:20]:
```suggestion
			urlParams.set('activepage', newPage);
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/usePagination.tsx [hunk 5]
```diff
@@ -5,9 +5,22 @@
...
+		}
+		setActivePage(newPage);
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/usePagination.tsx:22]:
```suggestion
		}

		setActivePage(newPage);
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/TableHeader/Search/Search.tsx [hunk 6]
```diff
@@ -25,12 +26,22 @@ const Search = ({initialSearchTerm, onSearchSubmit}: IProps) => {
 			onSearchSubmit(term);
 			setSearching(false);
 
+			if (term && urlParams) {
+				urlParams.set('searchterm', term);
+			}
+			else if (urlParams) {
+				urlParams.delete('searchterm');
+			}
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/TableHeader/Search/Search.tsx:34]:
```suggestion
            if (urlParams) {
                if (term) {
				    urlParams.set('searchterm', term);

                    return;
                }

                urlParams.delete('searchterm');
            }
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/fragments/claims-list-on-mdf-request-detail/index.js [hunk 7]
```diff
@@ -2,3 +2,16 @@
  * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
  * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
  */
+
+const viewButtons = document.getElementsByClassName('view-btn');
+
+if (viewButtons) {
+	const urlParams = new URLSearchParams(window.location.href.split('?')[1]);
+	for (let i = 0; i < viewButtons.length; i++) {
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/fragments/claims-list-on-mdf-request-detail/index.js:10]:
```suggestion

	for (let i = 0; i < viewButtons.length; i++) {
```

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/MDFRequestList.tsx (hunk 1 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/MDFRequestList.tsx]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/hooks/useFilters.tsx (hunk 2 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/hooks/useFilters.tsx]: Couldn't think in a efficient way to do this.

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/MDFRequestList.tsx (hunk 3 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/MDFRequestList.tsx]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/usePagination.tsx (hunk 4 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/usePagination.tsx]: It throws a type erro when i try to parse de number without casting to string.

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/usePagination.tsx (hunk 5 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/usePagination.tsx]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/TableHeader/Search/Search.tsx (hunk 6 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/TableHeader/Search/Search.tsx]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/fragments/claims-list-on-mdf-request-detail/index.js (hunk 7 above)
Comment (abfranca) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/fragments/claims-list-on-mdf-request-detail/index.js]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimList/hooks/useFilters.tsx [hunk 8]
```diff
@@ -72,6 +117,14 @@ export default function useFilters(
 			initialFilter = initialFilter
 				? initialFilter.concat(` and (${statusFilter})`)
 				: initialFilter.concat(`(${statusFilter})`);
+
+			urlParams.delete('status');
+			filters.status.value.forEach((value) =>
```
Comment (kylembischof) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimList/hooks/useFilters.tsx:122]: new line between `urlParams.delete` and `filters.status.value.forEach` as well

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimList/hooks/useFilters.tsx [hunk 9]
```diff
@@ -86,6 +139,14 @@ export default function useFilters(
 			initialFilter = initialFilter
 				? initialFilter.concat(` and (${partnerFilter})`)
 				: initialFilter.concat(`(${partnerFilter})`);
+
+			urlParams.delete('partner');
+			filters.partner.value.forEach((value) =>
```
Comment (kylembischof) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimList/hooks/useFilters.tsx:144]: new line

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimList/hooks/useFilters.tsx [hunk 10]
```diff
@@ -100,6 +161,14 @@ export default function useFilters(
 			initialFilter = initialFilter
 				? initialFilter.concat(` and (${partnerFilter})`)
 				: initialFilter.concat(`(${partnerFilter})`);
+
+			urlParams.delete('type');
+			filters.type.value.forEach((value) =>
```
Comment (kylembischof) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimList/hooks/useFilters.tsx:166]: new line

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/hooks/useFilters.tsx [hunk 11]
```diff
@@ -73,6 +118,14 @@ export default function useFilters(
 			initialFilter = initialFilter
 				? initialFilter.concat(` and (${statusFilter})`)
 				: initialFilter.concat(`(${statusFilter})`);
+
+			urlParams.delete('status');
+			filters.status.value.forEach((value) =>
```
Comment (kylembischof) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/hooks/useFilters.tsx:123]: new line

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/hooks/useFilters.tsx [hunk 12]
```diff
@@ -87,6 +140,14 @@ export default function useFilters(
 			initialFilter = initialFilter
 				? initialFilter.concat(` and (${partnerFilter})`)
 				: initialFilter.concat(`(${partnerFilter})`);
+
+			urlParams.delete('partner');
+			filters.partner.value.forEach((value) =>
```
Comment (kylembischof) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/hooks/useFilters.tsx:145]: new line

**Review by kylembischof — COMMENTED** [review]
hey felipe, in general when we are either declaring a variable and then executing some logic in an if statement, for loop, arrow function, etc. it is good practice to add a new line between them for easier readability. 
spacing out the variables and logic helps with that, which in turn will help with maintainability in the long run keeping our code readable and easy to work with

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/fragments/claims-list-on-mdf-request-detail/index.js [hunk 13]
```diff
@@ -2,3 +2,14 @@
  * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
  * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
  */
+
+const viewButtons = document.getElementsByClassName('view-btn');
+
+if (viewButtons) {
+	const urlParams = new URLSearchParams(window.location.search);
+	for (const viewButton of viewButtons) {
```
[the remark this answers is not in the document]
Comment (abfranca) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/fragments/claims-list-on-mdf-request-detail/index.js:11]: Hey, Kyle! Thanks for the advice!

[the remark this answers is not in the document]
Comment (abfranca) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/mdf-request-detail-collection/fragments/claims-list-on-mdf-request-detail/index.js:11]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/TableHeader/Search/Search.tsx [hunk 14]
```diff
@@ -25,12 +26,24 @@ const Search = ({initialSearchTerm, onSearchSubmit}: IProps) => {
...
+
 			return;
 		}
 
 		setTerm('');
 		onSearchSubmit('');
 		setSearching(true);
+		if (urlParams) {
```
[the remark this answers is not in the document]
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/TableHeader/Search/Search.tsx:46]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/useQueryParams.tsx [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+import {useEffect, useState} from 'react';
+
+import {Liferay} from '../services/liferay';
+
+export default function useQueryParams() {
+	const [urlParams] = useState(new URLSearchParams(window.location.search));
+	const urlParamsEntries = urlParams.entries();
+	useEffect(() => {
```
[the remark this answers is not in the document]
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/useQueryParams.tsx:14]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimList/hooks/useFilters.tsx (hunk 8 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimList/hooks/useFilters.tsx:123]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimList/hooks/useFilters.tsx (hunk 9 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimList/hooks/useFilters.tsx:146]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimList/hooks/useFilters.tsx (hunk 10 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimList/hooks/useFilters.tsx:169]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/hooks/useFilters.tsx (hunk 11 above)
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/hooks/useFilters.tsx:124]: Done!

**Comment by thaynam** [comment]
@abfranca add the Open/Completed tabs to the URL as well

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/TableHeader/Search/Search.tsx [hunk 16]
```diff
@@ -25,12 +26,26 @@ const Search = ({initialSearchTerm, onSearchSubmit}: IProps) => {
 			onSearchSubmit(term);
 			setSearching(false);
 
+			if (urlParams) {
+				if (term) {
+					urlParams.set('searchterm', term);
```
[the remark this answers is not in the document]
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/TableHeader/Search/Search.tsx]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/usePagination.tsx [hunk 17]
```diff
@@ -3,11 +3,25 @@
...
 
-import {useState} from 'react';
+import {useEffect, useState} from 'react';
 
-export default function usePagination() {
-	const [activeDelta, setActiveDelta] = useState<number>(20);
-	const [activePage, setActivePage] = useState<number>(1);
+export default function usePagination(urlParams?: URLSearchParams) {
+	const [activeDelta, setActiveDelta] = useState<number>(
+		urlParams?.get('activedelta')
```
[the remark this answers is not in the document]
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/usePagination.tsx]: Done!

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/usePagination.tsx [hunk 18]
```diff
@@ -3,11 +3,25 @@
...
+	);
+
+	const [activePage, setActivePage] = useState<number>(
+		urlParams?.get('activepage') ? Number(urlParams.get('activepage')) : 1
+	);
+
+	useEffect(() => {
+		if (urlParams) {
+			urlParams.set('activedelta', `${activeDelta}`);
+			urlParams.set('activepage', `${activePage}`);
```
[the remark this answers is not in the document]
Comment (abfranca) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/usePagination.tsx]: Done!

**Comment by abfranca** [comment]
ci:test:sf

**Review by thaynam — APPROVED** [review]

**Comment by thaynam** [comment]
ci:forward:force

**Comment by thaynam** [comment]
ci:forward:force

**Comment by thaynam** [comment]
ci:forward:force

**Pull request closed without merging**